### ISSUE TITLE
[Bug - FF / Safari] Lag while scrolling to top

### ISSUE THREAD

**Issue by aberonni** [abouolia/sticky-sidebar#6]
Steps to reproduce:

- Open https://abouolia.github.io/sticky-sidebar/
- Observe scrollable sticky element example
- Scroll to bottom of example
- Scroll slowly towards the top of the example
- Observe the sidebar

Here is a video that demonstrates the problem.
[scrollable-sticky-element.mov.zip](https://github.com/abouolia/sticky-sidebar/files/1074289/scrollable-sticky-element.mov.zip)

I'm looking at the code right now - if I find a fix I'll open a PR.

**Comment by aberonni** [comment]
It seems that it isn't an easy fix. I noticed that on FF the scroll event is triggered but the scrollTop isn't changed so sometimes the isScrollingTop function is returning a wrong value, still fixing that was just making things worse so I don't know what to say really.

**Comment by abouolia** [comment]
the lag when scrolling because of requestAnimationFrame function in https://github.com/abouolia/sticky-sidebar/blob/master/sticky-sidebar.js#L242 so i feel confused what the alternative.

**Comment by abouolia** [comment]
What is the browser used in the video? what it's version?

**Comment by degibons** [comment]
I noticed that bug too (FF 52-54)

**Comment by degibons** [comment]
Maybe this note will be useful for someones.
Sometimes (especially then the page have a lot of html and js) there is a bug with wrong sidebar container height calculation (was detected in FF and some other browsers).  
My solution in this case was to initiate `StickySidebar` with zero timeout:
```javascript
setTimeout(function(){
     $sidebar.stickySidebar(stickySidebarOptions);
},0);
```

**Comment by abouolia** [comment]
Tomorrow i will release a set of commits that solve this issue and other

**Comment by abouolia** [comment]
Hi @degibons @aberonni thanks for contribution and write about this issue, i fixed this issue and others , now sticky sidebar more smoother and performance, you can check it out. if there's any issue please don't hesitate to write about it. Regards.

**Comment by aberonni** [comment]
Nice one! I'll try it out ASAP and let you know.

**Comment by degibons** [comment]
Thanks, it really became more smoother in FF. But now, a warning message appears in the console:

------------------
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://developer.mozilla.org/docs/Mozilla/Performance/ScrollLinkedEffects for further details and to join the discussion on related tools and features!
------------------

Could the recent changes slow down the performance of the plugin?
Maybe it makes sense to leave the use of the RequestAnimationFrame, given that the global usage of FF less than 6.3% and in the Webkit browsers it works perfectly. What do you think?

**Comment by abouolia** [comment]
I've fixed all issues in version 2.0. check it out.

**Comment by degibons** [comment]
Can you tell what was the case, so I could fix this in the fisrt version of the plugin (which suits me more than the new one)?
